xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -1029,3 +1057,127 @@ DXGI_COLOR_SPACE_TYPE CProcessorHD::AvToDxgiColorSpace(const DXGIColorSpaceArgs&
 
   return DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P709;
 }
+
+bool CProcessorHD::IsSuperResolutionSuitable(const VideoPicture& picture)
+{
+  if (picture.iWidth > 1920)
+    return false;
+
+  const UINT outputWidth = DX::Windowing()->GetBackBuffer().GetWidth();
+
+  if (outputWidth <= picture.iWidth)
+    return false;
+
+  if (picture.iFlags & DVP_FLAG_INTERLACED)
+    return false;
+
+  if (picture.color_primaries == AVCOL_PRI_BT2020 ||
+      picture.color_transfer == AVCOL_TRC_SMPTE2084 ||
+      picture.color_transfer == AVCOL_TRC_ARIB_STD_B67)
+    return false;
+
```

Comment:
Is there a reason to exclude HDR?
1080p HDR exists and could need upscaling to 4k. UHD Bluray are not the only source of HDR.

Reply:
Yes, the reason is that not works with HDR 🙂

Reply:
OK seen in nVidia's FAQ, but Intel?